tests/test_webapi.py

```diff
@@ -0,0 +1,1084 @@
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import unittest
+import mock
+
+from nmoscommon.webapi import *
+
+from datetime import timedelta
+
+import flask
+
+class TestWebAPI(unittest.TestCase):
+
+    @mock.patch("nmoscommon.webapi.Flask")
+    @mock.patch("nmoscommon.webapi.Sockets")
+    def initialise_webapi_with_method_using_decorator(self, mock_webapi_method, decorator, Sockets, Flask, oauth_userid=None, expect_paths_below=False, is_socket=False, on_websocket_connect=None):
+        """Returns a pair of the wrapped function that would be sent to flask.app.route, and the parameters which would
+        be passed to flask.app.route for it."""
+        def TEST(self, *args, **kwargs):
+            return mock_webapi_method(*args, **kwargs)
+
+        class StubWebAPI(WebAPI):
+            pass
+
+        if on_websocket_connect is not None:
+            def _on_websocket_connect(*args, **kwargs):
+                return on_websocket_connect(*args, **kwargs)
+            StubWebAPI.on_websocket_connect = _on_websocket_connect
+
+        StubWebAPI.TEST = decorator(TEST)
+        basemethod = StubWebAPI.TEST
+
+        UUT = StubWebAPI()
+        if oauth_userid is not None:
+            UUT._oauth_config = { "loginserver" : "LOGINSERVER", "proxies" : { "http" : "HHTPPROXY", "https" : "HTTPSPROXY" }, 'access_whitelist' : [ oauth_userid, ] }
+        Flask.assert_called_once_with('nmoscommon.webapi')
+        app = Flask.return_value
+
+        self.assertEqual(app.response_class, IppResponse)
+        app.before_first_request.assert_called_once_with(mock.ANY)
+        torun = app.before_first_request.call_args[0][0]
+        Sockets.assert_called_once_with(app)
+
+        if is_socket:
+            Sockets.return_value.route.assert_called_once_with('/', endpoint="_TEST")
+            Sockets.return_value.route.return_value.assert_called_once_with(mock.ANY)
+            return (Sockets.return_value.route.return_value.call_args[0][0], UUT, [ (call[1], call[2]) for call in Sockets.return_value.route.mock_calls if call[0] == '' ])
+        elif not expect_paths_below:
+            app.route.assert_called_once_with('/',
+                                            endpoint="_TEST",
+                                            methods=mock.ANY)
+            app.route.return_value.assert_called_once_with(mock.ANY)
+        else:
+            self.assertItemsEqual((call for call in app.route.mock_calls if call[0] == ''), [
```

Comment:
Is this assertion being used for testing? If so it's not in a test, and if not then is it just meant to make the whole suite bomb out if it doesn't pass?